routes/products.js

```diff
@@ -20,18 +21,35 @@ routerProducts
   .route('/:productId')
   .get((req, res) => {
     const id = req.url.slice(1);
-    productDB.emptyMessage();
-    res.render(`templates/products/${productDB.retrieveOne(id).returnPage}`, productDB.retrieveOne(id));
+    const findProduct = productDB.getIndex(id);
+    if (findProduct === -1) {
+      return res.redirect('/products');
+    }
+    // if last message on specific product was for failed edit ...
+    // then remove the message when routing to "/products/:id" ...
+    // since message was displayed originally on re-route to "/products"
+    if (productDB.getMessage(id) === '{ "success": false, "message": "No field values were provided" }') {
```

Comment:
So what I like to do for things like this is to check if the message exists. If it does, pull that into a variable and remove the message from your db. Then just pass that variable into the res.render. This allows res.render to be the last night and you can prepend return to that to ensure nothing executes after that line.